build/configs/common_download.sh

```diff
@@ -0,0 +1,390 @@
+#!/usr/bin/env bash
+###########################################################################
+#
+# Copyright 2021 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+# common_download.sh
+
+CURDIR=$(readlink -f "$0")
+CUR_PATH=$(dirname "$CURDIR")
+BIN_PATH=${CUR_PATH}/../output/bin
+TOP_PATH=${CUR_PATH}/../..
+OS_PATH=${TOP_PATH}/os
+
+CONFIG=${OS_PATH}/.config
+source ${CONFIG}
+SMARTFS_BIN_PATH=${BIN_PATH}/${CONFIG_ARCH_BOARD}_smartfs.bin
+
+source ${CUR_PATH}/${CONFIG_ARCH_BOARD}/${CONFIG_ARCH_BOARD}_download.sh
+source ${CUR_PATH}/${CONFIG_ARCH_BOARD}/binary_name_map.txt
+
+## For manual port selection, set $3 as the portname, eg. make download ALL x ttyUSB2
+
+if [ -n "$3" ]; then
+	PORT="$3"
+else
+	PORT=${DEFAULT_PORT}
+fi
+
+WARNING="\n Port $PORT is selected\n\n
+	############################################\n
+	WARNINGS:\n
+	1. Make sure the board is in DOWNLOAD MODE.\n
+	2. Make sure NO other application like putty,\n
+	is occupying $PORT.\n
+	############################################\n"
+
+
+TTYDEV="/dev/${PORT}"
+
+##Utility function for sanity check##
+function sanity_check()
+{
+	if [ ! -f ${CONFIG} ];then
+		echo "No .config file"
+		exit 1
+	fi
+	
+	CHECK_BOARD=$(echo "CONFIG_ARCH_BOARD_${BOARD}")
+	if [[ "${!CHECK_BOARD}" != "y" ]];then
+		echo "Target is NOT ${BOARD}"
+		exit 1
+	fi
+
+	if fuser -s ${TTYDEV};then
+		echo "${TTYDEV} is used by another process, can't proceed"
+		exit 1
+	fi
+}
```

Comment:
Can we add a sanity check in some way to see if the board specific script and binary mapping files are present or not?